crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -1830,13 +1830,9 @@ impl OlmMachine {
     ) -> StoreResult<Vec<ExportedRoomKey>> {
         let mut exported = Vec::new();
 
-        let sessions: Vec<InboundGroupSession> = self
-            .store()
-            .get_inbound_group_sessions()
-            .await?
-            .into_iter()
-            .filter(|s| predicate(s))
-            .collect();
+        let mut sessions = self.store().get_inbound_group_sessions().await?;
+
+        sessions.retain(|s| predicate(s));
```

Comment:
(uber-nit) Maybe even `sessions.retain(predicate)` would work and be slightly more succinct.

Reply:
Nice spot, fixed in a fixup commit.